### PR TITLE
Ocr code

### PR DESCRIPTION
**Pull request by devshreebharatia**
The code implementation for optical Character Recognition UDF.
A unit test has also been added in test_pytorch.py to test the OCR UDF.

### CODE REVIEW

File: src/udfs/gpu_compatible.py [hunk 1]
```diff
@@ -13,9 +13,25 @@
...
+    def __init__(self):
+        self._device = NO_GPU
+
+    @property
+    def device(self,):
+        return self._device
+
+    def assign_device(self, device: str):
+        '''
+        Assigning the device passed by the UDF to the member variable of the class
```
Comment (gaurav274) [inline src/udfs/gpu_compatible.py:29]: We can be made clearer. Add comment to say it is an internal function. Thanks!

File: src/udfs/gpu_compatible.py [hunk 2]
```diff
@@ -13,9 +13,25 @@
...
+        Returns after enabling GPU for the device
+        '''
+        self._device = device
+        return self.to_device(device)
+
     @abstractmethod
```
Comment (gaurav274) [inline src/udfs/gpu_compatible.py:35]: Please also add a doc string for this function? Since it has to be implemented by the user, we should explicitly state the input and output requirements.

File: src/udfs/optical_character_recognition.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                    outcome = outcome.append(
+                        {
+                            "labels": list(pred_class),
+                            "scores": list(pred_score),
+                            "boxes": list(pred_boxes)
+                        },
+                        ignore_index=True)
+
+
+        return outcome
```
Comment (gaurav274) [inline src/udfs/optical_character_recognition.py:115]: Missing new line

File: src/udfs/optical_character_recognition.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+        """
+        
+        frames_list = frames.values.tolist()
+        frames = np.array(frames_list)
+
+        outcome = pd.DataFrame()
+        
+        for i in range(frames.shape[0]):
+            prediction = self.model.readtext_batched(frames[i])
```
Comment (gaurav274) [inline src/udfs/optical_character_recognition.py:74]: Isn't this logic processing one frame at a time? How about we call `self.model.readtext_batched(frames)`?

File: src/udfs/pytorch_abstract_udf.py [hunk 5]
```diff
@@ -38,8 +38,8 @@ def __init__(self):
         AbstractClassifierUDF.__init__(self)
         nn.Module.__init__(self)
 
-    def get_device(self):
-        return next(self.parameters()).device
+    #def get_device(self):
```
Comment (gaurav274) [inline src/udfs/pytorch_abstract_udf.py:41]: Are these changes required? I remember we decided to revert these back. Please double-check, thanks!

File: src/udfs/pytorch_abstract_udf.py [hunk 6]
```diff
@@ -52,7 +52,7 @@ def transform(self, images: np.ndarray):
 
     def forward(self, frames: List[np.ndarray]):
         tens_batch = torch.cat([self.transform(x) for x in frames])\
-            .to(self.get_device())
+            .to(f"cuda:{self.device}")
```
Comment (gaurav274) [inline src/udfs/pytorch_abstract_udf.py:55]: Similar as above.

**Review by gaurav274 — CHANGES_REQUESTED** [review]

File: src/udfs/pytorch_abstract_udf.py (hunk 5 above)
Comment (devshreebharatia) [reply src/udfs/pytorch_abstract_udf.py:41]: Yes, I think these changes are required. We are storing the device object using the assign_device() function in the GPU Compatible class as one of its variables. So we can directly access that object instead of using a get_device() function. I believe that was the reason we decided to eliminate this functionality. What do you think?

File: src/udfs/optical_character_recognition.py (hunk 4 above)
Comment (devshreebharatia) [reply src/udfs/optical_character_recognition.py:74]: I think there was some confusion due to the image size. I have made a few changes to make prediction on a batch of frames. Once we have the results for the entire batch, we traverse through each of the results and append them to the output dataframe. Creating a new PR for this!

**Comment by jarulraj** [comment]
Resolved in #303
